WIP : Write documentation (#29)
Joffref/genz: docs/testing.md

# Test your templates

## Why testing templates?

> Testing your templates is a good way to ensure that your templates are working as expected. 

Reviewer: ```suggestion
Testing your templates is a good way to ensure that:
* the generated code is exactly the one you expect
* the generated code does what you expect from it (by running it against unit tests)
```

> It's also a good way to ensure that your templates are not broken when you update GenZ or your templates themselves.

Reviewer: ```suggestion
It's also a good way catch regressions when you update GenZ or your templates themselves.
```

## How to test your templates?

GenZ provides a simple way to test your templates using the `genz test` command.

### Test your templates using the CLI

You can test your templates using the CLI with the following command:

```bash
genz test ./path/to/folder/containing/templates
```

> 

Reviewer: Should we explain that genz test is pluggable into any CI ?

Author: As it is just a binary to call, it is straight forward to me, but I might be wrong on this.
